python/cudf/cudf/core/column/numerical.py

```diff
@@ -280,6 +280,16 @@ def as_numerical_column(self, dtype: Dtype, **kwargs) -> NumericalColumn:
             return self
         return libcudf.unary.cast(self, dtype)
 
+    @property
+    def nan_count(self):
+        if self.dtype.kind != "f":
+            return 0
```

Comment:
Should we set `self._nan_cnt` to be zero in this case before we return?

Reply:
Yes, changed it